ql/src/java/org/apache/hadoop/hive/ql/security/authorization/PrivilegeSynchronizer.java

```diff
@@ -95,16 +103,27 @@ private void addACLsToBag(
                         (int) (System.currentTimeMillis() / 1000), GRANTOR, PrincipalType.USER, false), authorizer));
             break;
           case COLUMN:
-            privBag.addToPrivileges(
-                new HiveObjectPrivilege(new HiveObjectRef(HiveObjectType.COLUMN, dbName, tblName, null, columnName),
-                    principal, principalType, new PrivilegeGrantInfo(acl.getKey().toString(),
-                        (int) (System.currentTimeMillis() / 1000), GRANTOR, PrincipalType.USER, false), authorizer));
+
+            int privilegeBit = acl.getKey().ordinal();
+            columnPrivilegeBits[privilegeBit] = 1;
+            columnUpdateFlag = true;
+
             break;
           default:
             throw new RuntimeException("Get unknown object type " + objectType);
           }
         }
       }
+      if (columnUpdateFlag) {
+        String columnPrivilegeBitsString =
```

Comment:
is there any test case already present to test this ?

Reply:
InformationSchemaWithPrivilegeTestBase.java used by TestInformationSchemaWithPrivilegeZookeeperSSL.java  and TestInformationSchemaWithPrivilegeZookeeperPlain.java  , will initially add the acls to mock a table. At this time, the privilege synchroniser is called to update the privileges. If the privilege string was wrong, the test would have failed.